crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -357,6 +358,14 @@ void HKDF_verify_service_indicator(const EVP_MD *evp_md) {
   //
   // We can't test for that, as we don't know what the HKDF output will be
   // used for.
+  //
+  // Per SP800-56Crev2, the salt cannot be empty in FIPS mode; a NULL salt
+  // is replaced in the HMAC with a buffer filled with NUL bytes (0x00), so
+  // that's OK.
+  if (salt != NULL && salt_len == 0) {
```

Comment:
If I understand correctly, this function is once called with salt and once with info, does the same check apply to both? i.e. info cannot be empty in FIPS mode?
Also, HMAC with key=NULL (salt = NULL) is ok if md = NULL as in https://github.com/awslabs/aws-lc/blame/3f87d09e2b99e4523d60bce18b4fbfb0784f55e0/crypto/fipsmodule/hmac/hmac.c#L255. Can this occur in the calls in HMAC? I guess not. In that case, we may want to add to the comment that `evp_md` is always non-NULL.

Reply:
Based on what atsec said, the `info` argument in `HKDF_expand()` is the incoming salt when it's used in a stand-alone method. It's possible that this isn't quite right and will have to be adjusted once we start testing `HKDF_expand()` via ACVP.

Wow, nothing before this is checking for a `NULL` `md` so the `case` statement just under this would segfault...